### ISSUE TITLE
New Version 0.32. Query Model 'ModelConventionConnection' is already registered

### ISSUE THREAD

**Issue by rheid** [seekdavidlee/Eklee-Azure-Functions-GraphQl#64]
Hi David Lee, thank you very much for the update to the GraphQL version of .NET. I have started to switch to this version and have found a few problems. I am using two Querys variables with optional values. These are now throwing an exception in the new version.

            queryBuilderFactory.Create<Event>(coreQuery, "GetEvent")
            .WithPaging()
            .WithParameterBuilder()
            .WithProperty(x => x.Audiences, true)
            .WithProperty(x => x.Language, true)
            .WithProperty(x => x.Navigation, true)
            .WithProperty(x => x.MainTopics, true)
            .WithProperty(x => x.Name, true)
            .WithProperty(x => x.DateStart, true)
            .WithProperty(x => x.Place,true)
            .WithProperty(x => x.Abstract, true)
            .WithProperty(x => x.Description,true)

            .BeginQuery<Event>()

            .BuildQueryResult(ctx =>
            {
                var query = ctx.GetQueryResults<Event>();
                ctx.SetResults(query.OrderByDescending(q => q.CreatedOn).ToList());
            })
            .BuildQuery()
            .BuildWithListResult();

            queryBuilderFactory.Create<Page>(coreQuery, "GetPage")
                .WithPaging()
                .WithParameterBuilder()
                .WithProperty(x => x.Audiences, true)
                .WithProperty(x => x.Language, true)
                .WithProperty(x => x.Navigation, true)
                .WithProperty(x => x.MainTopics, true)
                .WithProperty(x => x.Name, true)
                .WithProperty(x => x.CreatedOn, true)
                .BeginQuery<Page>()

                .BuildQueryResult(ctx =>
                {
                    var query = ctx.GetQueryResults<Page>();
                    ctx.SetResults(query.OrderByDescending(q => q.CreatedOn).ToList());
                })
                .BuildQuery()
                .BuildWithListResult();

2021-01-24T17:08:49.819Z] Executed 'graphql' (Failed, Id=02497ed2-43ed-44cb-86f5-949e83684772, Duration=17089ms)
[2021-01-24T17:08:49.821Z] System.Private.CoreLib: Exception while executing function: graphql. GraphQL: Unable to register GraphType 'GraphQL.Types.Relay.ConnectionType`2[[Eklee.Azure.Functions.GraphQl.ModelConventionType`1[[genesis.com.api.Models.Page, genesis.com.api, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]], Eklee.Azure.Functions.GraphQl, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null],[GraphQL.Types.Relay.EdgeType`1[[Eklee.Azure.Functions.GraphQl.ModelConventionType`1[[genesis.com.api.Models.Page, genesis.com.api, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]], Eklee.Azure.Functions.GraphQl, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null]], GraphQL, Version=3.3.1.0, Culture=neutral, PublicKeyToken=null]]' with the name 'ModelConventionConnection';
the name 'ModelConventionConnection' is already registered to 'GraphQL.Types.Relay.ConnectionType`2[[Eklee.Azure.Functions.GraphQl.ModelConventionType`1[[genesis.com.api.Models.Event, genesis.com.api, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]], Eklee.Azure.Functions.GraphQl, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null],[GraphQL.Types.Relay.EdgeType`1[[Eklee.Azure.Functions.GraphQl.ModelConventionType`1[[genesis.com.api.Models.Event, genesis.com.api, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]], Eklee.Azure.Functions.GraphQl, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null]], GraphQL, Version=3.3.1.0, Culture=neutral, PublicKeyToken=null]]'.

**Comment by seekdavidlee** [comment]
Hi @rheid  - This is a known issue with GraphQL.NET library where I have reported it in https://github.com/graphql-dotnet/relay/issues/72. In the meantime, the workaround is to create a separate endpoint to host your paging implementation.

See documentation: https://github.com/seekdavidlee/Eklee-Azure-Functions-GraphQl/blob/master/Documentation/Queries.md#paging-implementation

**Comment by rheid** [comment]
@seekdavidlee  Thank you for your super quick reply. I am quite sure that I have understood this correctly or that I can implement it. Do you have an example? Should these endpoints be implemented again under a different name for the paging?

`public static class GraphQLFunction
    {
        [ExecutionContextDependencyInjection(typeof(FunctionModule))]
        [FunctionName("graphql")]
        public static async Task<IActionResult> Run(
            [HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)] HttpRequest req,
            ILogger log,
            ExecutionContext executionContext)
        {
            return await executionContext.ProcessGraphQlRequest(req);
        }
    }`

A small code example would help.

**Comment by seekdavidlee** [comment]
Yes @rheid - You can refer to the example project where in my same project Eklee.Azure.Functions.GraphQl.Example, I have multiple endpoints. You would create a new Azure Function, new AutoFac Module, new Mutation, new Query specifically for each paging Model you are exposing. Via dependency injection, they are seen as different instances, and so the error from GraphQL.NET wouldn't surface.

![image](https://user-images.githubusercontent.com/36802330/105645272-355b9d80-5e60-11eb-87e6-5dca3b362d74.png)

**Comment by rheid** [comment]
@seekdavidlee  I'll have a look at that tomorrow. thank you very much.
